test_app/after/src/macros_in_binary.erl

```diff
@@ -0,0 +1,16 @@
+-module(macros_in_binary).
+
+-export([chunkify/1, p/0]).
+
+-define(CHUNK_SIZE, 65535 - 1).
+-define(CONCAT(A, B), <<A/binary, B/binary>>).
+-define(A_MACRO(X), X + 1).
+
+chunkify(<<Binary:(?CHUNK_SIZE)/binary, Rest/binary>>) ->
+    {<<(?CONCAT(Binary, Rest))/binary>>, <<(?CONCAT(Binary, Rest))/binary>>}.
+
+p() ->
+    <<begin
+          X = rand:uniform(10),
+          X + (?A_MACRO(X))
```

Comment:
This does not match the way this same code would be formatted anywhere else in the code. It should not have parentheses in this case.

Reply:
Yep, but this code if safe. I totally agree with that we need to expand the logic for check if this mathematical expression or not in binary and also do the same check if macros have the mathematical expression, but my main goal was to fix the error during formatting, since the code from the wizard can break the code, while this PR fixes the current problem. So, as I provide in description of PR `A little bit dirty but working solution.` :upside_down_face:.

Reply:
At the same time we can continue working on this PC, in separate commit. So, do you have any approaches about it? I see it this way, first we check or the last expression in the macro is mathematical, if yes, we check or in binaries where this macro is used, a mathematical expression is used, and on the basis of this we can already build logic. How this sounds for you?

Reply:
Well... My gut feeling (didn't test it) would be to approach the problem from a different angle.
The formatter is likely adding parentheses already on things like `<<(a_func(Var))/binary>>`.
We just need it to treat a macro application in the same way as that function application.
So... Instead of adding a "temporary" field to `#ctxt`, I would try to see if I can _persuade_ the formatter ;)

Reply:
Ok, I will continue to research, if you get new ideas or approaches, feel free to share them here or send them to me at v.katsuba.dev@gmail.com. For now I will change title of PR and will add prefix **WIP**.

Reply:
Adding parentheses around macros is not always safe. See:

```erl
-define(A_MACRO(X), X + 2).

2 * ?A_MACRO(2) %% is 6
2 * (?A_MACRO(2)) %% is 8
```
```

Reply:
Yeah. And we have another ticket for that. <I'll put the ticket # here when I get back to my computer>